database/nakadi/10_data/04_tables/03_event_type_schema.sql

```diff
@@ -0,0 +1,9 @@
+CREATE TABLE IF NOT EXISTS zn_data.event_type_schema (
+  ets_event_type_name varchar(255) NOT NULL REFERENCES zn_data.event_type (et_name),
+  ets_schema_object jsonb NOT NULL
+);
+
+CREATE INDEX ON zn_data.event_type_schema USING gin (ets_schema_object);
+
+CREATE INDEX ON zn_data.event_type_schema ((ets_schema_object->>'version'));
+CREATE INDEX ON zn_data.event_type_schema ((ets_schema_object->>'created_at'));
```

Comment:
I think you can also create unique index for pair (event-type-name, version). That will guarantee the uniqueness of that pair from DB point of view.